lib/unique_user_events.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require 'unique_user_events/service'
+
+# Top-level module for Unique User Metrics system
+#
+# This module provides the main entry point for logging unique user events
+# for analytics purposes. Events are tracked only once per user and include
+# automatic DataDog metric increments for new events.
+#
+# @example Basic usage
+#   UniqueUserEvents.log_event(user_id: current_user.uuid, event_name: 'mhv_landing_page_accessed')
+#
+# @example Check if event exists
+#   UniqueUserEvents.event_logged?(user_id: current_user.uuid, event_name: 'secure_messages_accessed')
+module UniqueUserEvents
+  # Log a unique user event with DataDog metrics
+  #
+  # This method records the event in the database (if new) and increments
+  # a DataDog counter for unique user metrics analytics.
+  #
+  # @param user_id [String] UUID of the authenticated user
+  # @param event_name [String] Name of the event being logged (max 50 chars)
+  # @return [Boolean] true if new event was logged, false if already existed or feature disabled
+  # @raise [ArgumentError] if parameters are invalid
+  #
+  # @example
+  #   UniqueUserEvents.log_event(user_id: user.uuid, event_name: 'appointments_viewed')
+  def self.log_event(user_id:, event_name:)
+    return false unless Flipper.enabled?(:unique_user_metrics_logging)
+
+    start_time = Time.current
+    result = Service.log_event(user_id:, event_name:)
+    duration = (Time.current - start_time) * 1000.0 # Convert to milliseconds
+
+    StatsD.measure('uum.unique_user_metrics.log_event.duration', duration, tags: ["event_name:#{event_name}"])
```

Comment:
Note: We will use this to monitor latency of this operation. We are expecting the database call to take longer and longer as the table grows, but we have optimized that table to the point that the latency should be minimal. In any case, we can monitor it with this.